example_test.go

```diff
@@ -85,6 +85,36 @@ func ExampleRepository_References() {
 
 }
 
+func ExampleRepository_ResolveRevision() {
+	r, _ := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
+		URL: "https://github.com/git-fixtures/basic.git",
+	})
+
+	// Resolve revision to their corresponding sha
+	revisions := []string {
+		"master~1",
+		"HEAD~2^^~",
+		"HEAD~2^{/binary file}",
+		"refs/heads/master~2^^~",
+	}
+
+	for _, revision := range revisions {
+		h, err := r.ResolveRevision(plumbing.Revision(revision))
+
+		if err != nil {
+			log.Fatalf("%s : %s", revision, err)
+		}
+
+		fmt.Printf("%s  %s\n", h.String(), revision)
+	}
+
+	// Output:
```

Comment:
I saw in others example, you have `// Example Output` instead of `// Output`, so examples are not compiled, is it a mistake or something you explicitly did ?

Reply:
If the examples are in _examples folder are tested in a different way. But if is a example in other folder is an error